### PR TITLE
Update Java version to 11 Part 1 (#617 / #627)

### PR DESCRIPTION
**Pull request by Bukama**
Proposed commit message:

```
Starting with release 2.0 Pioneer sets Java 11 as it's base Java version.
With this PR only the build is updated to 11 without any further code changes to create the new base first.

related #617
PR: #627
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Check out [junit-pioneer.org](https://junit-pioneer.org/), particularly [the doc
 ## A Pioneer's Mission
 
 JUnit Pioneer provides extensions for [JUnit 5](https://github.com/junit-team/junit5/) and its Jupiter API.
-It does not limit itself to proven ideas with wide application but is purposely open to experiments.
+It does not limit itself to proven ideas with wide application but is purposely open to experiment.
```
Comment (Michael1993) [inline README.md:19]:
```suggestion
It does not limit itself to proven ideas with wide application but is purposely open to experiments.
```

File: build.gradle.kts [hunk 2]
```diff
@@ -114,15 +108,15 @@ sonarqube {
...
-moditect {
-	addMainModuleInfo {
-		version = project.version
-		overwriteExistingFiles.set(true)
-		module {
-			moduleInfoFile = rootProject.file("src/main/module/module-info.java")
-		}
-	}
-}
+//moditect {
```
Comment (Michael1993) [inline build.gradle.kts:111]: Don't comment code out - remove it.

File: src/main/java/module-info.java [hunk 3]
```diff
@@ -3,13 +3,15 @@
  * and its Jupiter API.
  *
  * <p>Pioneer does not limit itself to proven ideas with wide application but is purposely open to
- * experiments. It aims to spin off successful and cohesive portions into sibling projects or back
+ * experiment. It aims to spin off successful and cohesive portions into sibling projects or back
```
Comment (Michael1993) [inline src/main/java/module-info.java:6]:
```suggestion
 * experiments. It aims to spin off successful and cohesive portions into sibling projects or back
```

File: src/main/java/module-info.java [hunk 4]
```diff
@@ -3,13 +3,15 @@
...
  * into the JUnit 5 code base.
  *
  * <p>The dependencies on Jupiter modules could be marked as <code>transitive</code> but that would
  * allow users who depend on this module to not `require` org.junit.*, which would be backwards.
  */
 module org.junitpioneer {
+	requires java.se;
```
Comment (Michael1993) [inline src/main/java/module-info.java:13]: Is this necessary?

File: README.md (hunk 1 above)
Comment (Bukama) [reply README.md:19]: We should come to a a common if it should be an noun or a verb :D

File: src/main/java/module-info.java (hunk 4 above)
Comment (Bukama) [reply src/main/java/module-info.java:13]: Yes otherwise the build fails for me, missing the `java.util` things.

File: build.gradle.kts (hunk 2 above)
Comment (Bukama) [reply build.gradle.kts:111]: Good spot

File: README.md (hunk 1 above)
Comment (Bukama) [reply README.md:19]: We decided in Discord to change it to "but is purposely open to experimentation"

File: src/main/java/module-info.java (hunk 4 above)
Comment (Michael1993) [reply src/main/java/module-info.java:13]: Okay. One of the nuances of the module system, I guess.

File: src/main/java/module-info.java [hunk 5]
```diff
@@ -19,6 +21,7 @@
 	requires static com.fasterxml.jackson.core;
 	requires static com.fasterxml.jackson.databind;
 
+
```
Comment (Michael1993) [inline src/main/java/module-info.java:24]: What?! An unnecessary extra empty line? Blasphemy! Scandalous!

File: build.gradle.kts [hunk 6]
```diff
@@ -23,23 +22,18 @@ plugins.withType<JavaPlugin>().configureEach {
 group = "org.junit-pioneer"
 description = "JUnit 5 Extension Pack"
 
-val modularBuild : String by project
 val experimentalJavaVersion : String? by project
 val experimentalBuild: Boolean = experimentalJavaVersion?.isNotEmpty() ?: false
 
-val targetJavaVersion = JavaVersion.VERSION_1_8
+val targetJavaVersion = JavaVersion.VERSION_11
```
Comment (Michael1993) [inline build.gradle.kts:28]: Please also fix line 211 (--release option)

**Review by Michael1993 — APPROVED** [review]
Looks fine to me. We want this as part of 2.0, right?

**Comment by Bukama** [comment]
Adding `merge-ready` to check full set of jobs.

**Review by aepfli — APPROVED** [review]
lgtm

**Comment by nipafx** [comment]
image
Eeek!

**Comment by nipafx** [comment]
There was a weird merge in there that I removed and now the diff looks better.
image

**Comment by nipafx** [comment]
PS: That means I force pushed! 😈

**Comment by Bukama** [comment]
Blocked until actual release of Pioneer 2.0

**Review by beatngu13 — APPROVED** [review]

**Comment by beatngu13** [comment]
@Bukama did the rebase and LGTM. As the original author, could you please have another look and if everything's fine, proceed with the merge into `lab/2.0`?

**Comment by Bukama** [comment]
> @Bukama did the rebase and LGTM. As the original author, could you please have another look and if everything's fine, proceed with the merge into `lab/2.0`?

Looks good to me. As discussed in chat, we both have no clue, why the build is not triggered and decided to merge (as there are no conflicts) and then check the `lab/2.0` branch.

**Pull request merged**